public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ConfigEditor.tsx

```diff
@@ -33,10 +34,8 @@ export class ConfigEditor extends PureComponent<Props, State> {
       unsaved: false,
       appInsightsInitiallyConfigured: isAppInsightsConfigured(props.options),
     };
-
-    if (this.props.options.id) {
-      updateDatasourcePluginOption(this.props, 'url', '/api/datasources/proxy/' + this.props.options.id);
-    }
+    // TODO: Can I get resources from here?
+    this.baseURL = `/api/datasources/${this.props.options.id}/resources/${routeNames.azureMonitor}/subscriptions`;
```

Comment:
is there a way to use `getResource` or similar here @sunker? (or do I need to use the raw URL like I am doing)

Reply:
hmm. I think instead of having this file do backend requests, it should be using the data source file(s) to communicate with the backend. Inside the config editor, you should be able to load the ds async doing something like this:
```typescript
getDatasourceSrv().loadDatasource(this.props.options.name)
```
then we can, for example, use the getSubscriptions method inside the datasource file instead of re-implementing that in this file. 

just beware that the the first time you enter the config page, you don't have a data source yet. it will be created once you click Save and test, so you need to guard for that.

Reply:
the problem with that then would be that we need to load subscriptions _before_ hitting "Save and test". Note that for example the field "Default Subscription" is necessary, we cannot expect users to hit the save button and then continue to edit the form (and hit save again).

Reply:
I see. but how did that work before? was the pluginproxy available to the plugin also before "Save and test" was clicked?

Reply:
oooh, the button "Load Subscriptions" does the trick... It's actually saving the datasource and then doing the request (I thought it was automatic, not sure how tbh). I guess we can rely in the same procedure, let me give that a try

Reply:
:+1:

Reply:
This is not working for me. `getDatasourceSrv().loadDatasource` is returning a "not found" error while `await getBackendSrv().get(/api/datasources/${this.props.options.id})` actually works. I am not sure what `getDatasourceSrv()` is using as storage? I don't see a way to refresh that state (if I refresh the page `getDatasourceSrv().loadDatasource` returns the proper datasource).